backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/KonanDriver.kt

```diff
@@ -43,7 +43,7 @@ fun runTopLevelPhases(konanConfig: KonanConfig, environment: KotlinCoreEnvironme
         KonanPhases.list()
     }
 
-    if (config.kotlinSourceRoots.isEmpty()) return
+    if (config.kotlinSourceRoots.isEmpty() && !konanConfig.linkOnly) return
```

Comment:
`if (!config.enoughArguments) return`?

Reply:
Agree, but I think it's better to use another name, say `runPhases` or `infoArgsOnly`.

I mean technically `enoughArguments` should be true if something like `-list_targets` or `--list_phases` is specified. But this line checks if we need to execute the compiler phases (there are source files to compile and/or libraries to link) so it should return also if e.g. `-list_targets`  is passed (thus `enoughArguments` is true) but other args aren't.